src/util.c

```diff
@@ -145,6 +147,92 @@ int hyper_find_sd(char *addr, char **dev)
 	return 0;
 }
 
+static unsigned long id_or_max(const char *name)
+{
+	char *ptr;
+	long id = strtol(name, &ptr, 10);
+	if (name == ptr || id < 0 || (errno != 0 && id == 0) || *ptr != '\0')
```

Comment:
when there is any error,  strtol return 0. this test is copied from the example in man pages of strtol.
```
           if ((errno == ERANGE && (val == LONG_MAX || val == LONG_MIN))
                   || (errno != 0 && val == 0)) {
               perror("strtol");
               exit(EXIT_FAILURE);
           }
```